src/Interface/Modules/Render/ViewScene.cc

```diff
@@ -1354,7 +1381,8 @@ void ViewSceneDialog::selectObject(const int x, const int y)
 
     if (selectedWidget_)
     {
-      colorWidgetRed(selectedWidget_);
+      widgetColorChanger_ = boost::make_shared<ScopedWidgetColorChanger>(selectedWidget_);
```

Comment:
Here's actually a better way to structure the logic: create the scoped object and pass the `selectedWidget_` in all the time. The scoped object will then test for null and change the color if not null. Then the destructor resets the color if the not null again. That can eliminate the if statements in this function, and the need to allocate the scoped object conditionally